Assets/BossRoom/Scripts/Client/UI/Lobby/LobbyUIMediator.cs

```diff
@@ -0,0 +1,338 @@
+
+        private LobbyAsyncRequests m_LobbyAsyncRequests;
+        private LobbyUser m_localUser;
+        private LocalLobby m_localLobby;
+        private LobbyServiceData m_lobbyServiceData;
+        private LobbyContentHeartbeat m_lobbyContentHeartbeat;
+        private IPublisher<UnityServiceErrorMessage> m_UnityServiceErrorMessagePublisher;
+        private NameGenerationData m_NameGenerationData;
+        private GameNetPortal m_GameNetPortal;
+        private ClientGameNetPortal m_ClientNetPortal;
+
+        [Inject]
+        private void InjectDependencies(
+            LobbyAsyncRequests lobbyAsyncRequests,
+            IPublisher<UnityServiceErrorMessage> unityServiceErrorMessagePublisher,
+            LobbyUser localUser,
+            LobbyContentHeartbeat lobbyContentHeartbeat,
+            LobbyServiceData lobbyServiceData,
+            LocalLobby localLobby,
+            NameGenerationData nameGenerationData,
+            GameNetPortal gameNetPortal,
+            ClientGameNetPortal clientGameNetPortal
+        )
+        {
+            //m_persistentPlayer = persistentPlayer;
+
+            m_NameGenerationData = nameGenerationData;
+            m_localUser = localUser;
+            m_LobbyAsyncRequests = lobbyAsyncRequests;
+            m_UnityServiceErrorMessagePublisher = unityServiceErrorMessagePublisher;
+            m_lobbyContentHeartbeat = lobbyContentHeartbeat;
+            m_lobbyServiceData = lobbyServiceData;
+            m_localLobby = localLobby;
+            m_GameNetPortal = gameNetPortal;
+            m_ClientNetPortal = clientGameNetPortal;
+
+            RegenerateName();
+
+            m_ClientNetPortal.NetworkTimedOut += OnNetworkTimeout;
+            m_ClientNetPortal.ConnectFinished += OnConnectFinished;
+
+            //any disconnect reason set? Show it to the user here.
+            ConnectStatusToMessage(m_ClientNetPortal.DisconnectReason.Reason, false);
+            m_ClientNetPortal.DisconnectReason.Clear();
+        }
+
+        private void OnDestroy()
+        {
+            if (m_ClientNetPortal != null)
+            {
+                m_ClientNetPortal.NetworkTimedOut -= OnNetworkTimeout;
+                m_ClientNetPortal.ConnectFinished -= OnConnectFinished;
+            }
+        }
+
+        //Lobby and Relay calls done from UI
+
+        public void CreateLobbyRequest(string lobbyName, bool isPrivate, int maxPlayers, OnlineMode onlineMode, string ip, int port)
+        {
+            m_LobbyAsyncRequests.CreateLobbyAsync(lobbyName, maxPlayers, isPrivate, onlineMode, ip, port, OnCreatedLobby, OnFailedLobbyCreateOrJoin);
+            BlockUIWhileLoadingIsInProgress();
+        }
+
+        public void QueryLobbiesRequest(bool blockUI)
+        {
+
+            m_LobbyAsyncRequests.RetrieveLobbyListAsync(
+                OnSuccess,
+                OnFailure
+            );
+
+            if (blockUI)
+            {
+                BlockUIWhileLoadingIsInProgress();
```

Comment:
Any reason why we don't a generic reusable throughout boss room "loading" overlay?

Reply:
(doesn't have to be addressed in this PR, can be a next step)